### DISCUSSION TITLE
Ratelimit hash (zbase32?)

### DISCUSSION THREAD

**Discussion by dominykasn** [rspamd/rspamd#4178] [Q&A]
Hi, fellow rspamd users,
is it in any way possible to generate the rspamd_hash:create():base32 hash outside rspamd application? I've tried several encoding applications including zbase32 encoding and could not achieve the given function hash below:
```
-- /usr/share/rspamd/plugins/ratelimit.lua
rspamd_hash.create(redis_key):base32()
```

**Comment by fatalbanana** [comment]
What language do you hope to use? Rspamd's zbase32 encoding is non-standard for legacy reasons; we have implementations in [Rust](https://github.com/vstakhov/rust-base32) and [Go](https://github.com/vstakhov/go-base32) so far.
